docs/plugin-development/plugin-metadata.md

```diff
@@ -0,0 +1,104 @@
+# Setting Plugin Metadata
+
+Writing a plugin is all well and good, but you're also going to want some way to
+configure how it appears in the plugin installer: its name, description, icon,
+update changelogs, and so on. You do this by writing a _plugin manifest_, which
+is a JSON or YAML file named after your plugin's internal name (for example:
+`SamplePlugin.yaml`).
+
+:::info
+
+When using a YAML manifest, you should replace CamelCase keys with snake_case
+(for example, where a JSON manifest would contain `RepoUrl`, a YAML manifest
+would contain `repo_url`).
+
+:::
+
+## Example
+
+A very basic YAML manifest might look something like this:
+
+```yaml
+name: Test Plugin
+author: You
+punchline: Does nothing! # short summary that fits on one line
+description: |- # long description, shown when your plugin is clicked on
+  This is a test plugin - this first line is a summary.
+
+  Down here is a more detailed explanation of what the plugin
+  does, manually wrapped to make sure it stays visible in the
+  installer.
+repo_url: https://example.com # where can users find your plugin's source code?
+```
+
+## Available manifest keys
+
+These keys are required:
+
+- Name
+- Author
+- Description
+- Punchline
+
+Optional keys include:
+
+- ApplicableVersion
+- RepoUrl
+- Tags
+- CategoryTags
+- LoadRequiredState
+- LoadSync
+- CanUnloadAsync
+- LoadPriority
+- ImageUrls
+- IconUrl
+- Changelog (see [Changelogs](#changelogs) below for alternatives)
+- AcceptsFeedback
+- FeedbackMessage
+
+See [the definition of `IPluginManifest`][] for further information. NB: some
+fields mentioned there, like `Dip17Channel`, are set automatically by the
+various plumbing that gets your plugins from GitHub to people's computers, and
+you should not include them in your manifest explicitly. (For a list of the keys
+you're allowed to set, see [the relevant part of DalamudPackager][].)
+
+Note that the following keys are also required for Dalamud to load a plugin, but
+if you're using [DalamudPackager][], it will automatically fill them in for you,
+so **you can and should ignore these**:
+
+- AssemblyVersion
+- InternalName
+- DalamudApiLevel
+
+## Changelogs
+
+You can set a plugin changelog in three separate ways:
+
+1. Include a `changelog` field in your `manifest.toml`, in the
+   [DalamudPlugins17][] repository
+2. Write text in your Pull Request description
+3. Include the `Changelog` key in your plugin manifest
+
+These will be checked in order, and the first available changelog will be used.
+
+:::warning
+
+If you'd like your changelog to be visible in the plugin installer, you must
```

Comment:
This isn't accurate, changelogs in PR descriptions will also show up in the installer. At the very least they will be in the changelogs tab, if they're not in the installed plugins menu, that would be a bug - should be a simple backend change though.

Reply:
Turns out the situation is more complex than either of us thought.

Take Daily Duty v4.0.6.1 (https://github.com/goatcorp/DalamudPluginsD17/pull/2930) as an example. There's no changelog in manifest.toml or in the plugin manifest, but there is a PR description, and it does not contain `nofranz`.

- It was, correctly, [posted in Discord](https://discord.com/channels/581875019861328007/663675584844660736/1177489131601940561)
- It does _not_ show up in the installed plugins tab
  - I think this is because its PluginMaster entry has null `Changelog`
- It does _not_ show up in my changelog tab
  - and yet https://kamori.goats.dev/Plugin/History/DailyDuty?track=stable shows it!
  - the last Daily Duty changelog I see is from 4.0.3.1, which matches https://kamori.goats.dev/Plugin/History/DailyDuty?track=testing-live
  - I'm not using Daily Duty from testing ("receive plugin testing versions" unchecked), so this seems like a bug
  - presumed cause: the PluginMaster entry has `"_Dip17Channel": "testing-live"`, and it seems like DalamudChangelogManager uses that to figure out which changelogs to pull https://github.com/goatcorp/Dalamud/blob/22d27fbb192d2d9b746853cfe376b01c1ebb6492/Dalamud/Interface/Internal/Windows/PluginInstaller/DalamudChangelogManager.cs#L56-L59